File: mvcc/record.go
package mvcc

type Record struct {
	id       string // unique identifier for the record
    vector   []float64
    metadata map[string]string
    beginTS  int  // begin timestamp
    endTS    int  // end timestamp (math.MaxInt32 if still valid)
    deleted  bool // true if deleted
	createdByTxnID    int  // transaction ID that created this version
    next     *Record // newer version
}

File: mvcc/record.py
import math

class Record:
    def __init__(self, id: str, value: str = ""):
        self.id = id
        self.key = ""
        self.begin_ts: int | float = 0
        self.end_ts: int | float = math.inf
        self.deleted: bool = False
        self.created_by_txn_id: int | None = None
        # store the raw text and its vector
        self.value: str = value
        self.next: Record | None = None